MixStaticSeq/training.py

```diff
@@ -165,27 +182,27 @@ def _init_new_model(self):
                                      nonseq_output_dim=self.args.nonseq_representation_dim,
                                      mix_output_dim=self.args.mix_output_dim,
                                      loss_mode=self.args.loss_mode)
+        self.config.sampling_weight = self.args.sampling_weight
         self.model = MixModel(config=self.config, model_type=self.args.model_type)
         self.model.to(self.args.device)
 
         # set up optimizer
         if self.args.optim == ModelOptimizers.ADAM.value:
             # using AdamW implementation
             # may have problem with batch=1 since it is more fit for mini-batch update
-            # TODO may add no_decay params
-            # no_decay = ['bias']
-            # optimizer_grouped_parameters = [
-            #     {'params': [p for n, p in model.named_parameters() if not any(nd in n for nd in no_decay)],
-            #      'weight_decay': args.weight_decay},
-            #     {'params': [p for n, p in model.named_parameters() if any(nd in n for nd in no_decay)],
-            #      'weight_decay': 0.0}
-            # ]
-            self.optimizer = torch.optim.AdamW(self.model.parameters(), lr=self.args.learning_rate, amsgrad=True)
+            no_decay = {'bias'}
+            optimizer_grouped_parameters = [
+                {'params': [p for n, p in self.model.named_parameters() if not any(nd in n for nd in no_decay)],
+                 'weight_decay': self.args.weight_decay},
+                {'params': [p for n, p in self.model.named_parameters() if any(nd in n for nd in no_decay)],
+                 'weight_decay': 0.0}
+            ]
+            self.optimizer = torch.optim.AdamW(optimizer_grouped_parameters, lr=self.args.learning_rate, amsgrad=True)
         elif self.args.optim == ModelOptimizers.SGD.value:
             # using momentum SGD implementation and default momentum is set to 0.9 and use nesterov
             # high variance of the parameter updates; less stable convergence; but work with batch=1
-            self.optimizer = torch.optim.SGD(self.model.parameters(), lr=self.args.learning_rate,
-                                             momentum=0.9, nesterov=True)
```

Comment:
Should momentum always equal to 0.9?

Reply:
0.9 is the best practice; in general, we do not need to tune it.